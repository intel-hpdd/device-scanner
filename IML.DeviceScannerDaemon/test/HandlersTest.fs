// Copyright (c) 2018 Intel Corporation. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

module IML.DeviceScannerDaemon.HandlersTest

open Handlers
open TestFixtures
open Fable.Import.Jest
open Matchers
open IML.Types.CommandTypes

testList "Data Handler" [
  let withSetup f ():unit =
    f (backCompatHandler)

  yield! testFixture withSetup [
    "Should call end with map for info event", fun (handler) ->
      handler infoUdev
        |> toMatchSnapshot

    "Should add then remove a device path", fun (handler) ->
      expect.assertions 2
      handler (UdevCommand addUdev) |> toMatchSnapshot

      handler (UdevCommand removeUdev) |> toMatchSnapshot
    ]
]
