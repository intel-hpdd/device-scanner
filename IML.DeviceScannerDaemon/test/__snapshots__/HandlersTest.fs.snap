// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Handler Should add then remove a device path 1`] = `
Data {
  "blockDevices": FableMap {
    "comparer": Comparer {
      "Compare": [Function],
    },
    "tree": MapTree {
      "data": Array [
        "/devices/pci0000:00/0000:00:01.1/ata1/host1/target1:0:0/1:0:0:0/block/sdb/sdb1",
        UEvent {
          "DEVNAME": "/dev/sdb1",
          "DEVPATH": "/devices/pci0000:00/0000:00:01.1/ata1/host1/target1:0:0/1:0:0:0/block/sdb/sdb1",
          "DEVTYPE": "partition",
          "DM_LV_NAME": "swap",
          "DM_MULTIPATH_DEVICE_PATH": true,
          "DM_UUID": "LVM-pV8TgNKMJVNrolJgMhVwg4CAeFFAIMC83IU1hvimWWlvmd5xQddtMIqRtjwOuKTz",
          "DM_VG_NAME": "centos",
          "ID_FS_TYPE": "LVM2_member",
          "ID_FS_USAGE": "filesystem",
          "ID_MODEL": "VBOX_HARDDISK",
          "ID_PART_ENTRY_NUMBER": 1,
          "ID_SERIAL": "VBOX_HARDDISK_VB304a0a0f-15e93f07",
          "ID_VENDOR": null,
          "IML_DM_SLAVE_MMS": Array [],
          "IML_DM_VG_SIZE": null,
          "IML_IS_RO": true,
          "IML_MD_DEVICES": Array [],
          "IML_SCSI_80": "80",
          "IML_SCSI_83": "83",
          "IML_SIZE": "81784832",
          "MAJOR": "8",
          "MD_UUID": null,
          "MINOR": "17",
          "PATHS": Array [
            "/dev/sdb1",
            "/dev/disk/by-id/ata-VBOX_HARDDISK_VB304a0a0f-15e93f07-part1",
            "/dev/disk/by-path/pci-0000:00:01.1-ata-1.0-part1",
          ],
        },
      ],
      "tag": 1,
    },
  },
  "zed": ZedData {
    "props": FableSet {
      "comparer": Comparer {
        "Compare": [Function],
      },
      "tree": SetTree {
        "data": undefined,
        "tag": 0,
      },
    },
    "zfs": FableSet {
      "comparer": Comparer {
        "Compare": [Function],
      },
      "tree": SetTree {
        "data": undefined,
        "tag": 0,
      },
    },
    "zpools": FableMap {
      "comparer": Comparer {
        "Compare": [Function],
      },
      "tree": MapTree {
        "data": undefined,
        "tag": 0,
      },
    },
  },
}
`;

exports[`Data Handler Should add then remove a device path 2`] = `
Data {
  "blockDevices": FableMap {
    "comparer": Comparer {
      "Compare": [Function],
    },
    "tree": MapTree {
      "data": undefined,
      "tag": 0,
    },
  },
  "zed": ZedData {
    "props": FableSet {
      "comparer": Comparer {
        "Compare": [Function],
      },
      "tree": SetTree {
        "data": undefined,
        "tag": 0,
      },
    },
    "zfs": FableSet {
      "comparer": Comparer {
        "Compare": [Function],
      },
      "tree": SetTree {
        "data": undefined,
        "tag": 0,
      },
    },
    "zpools": FableMap {
      "comparer": Comparer {
        "Compare": [Function],
      },
      "tree": MapTree {
        "data": undefined,
        "tag": 0,
      },
    },
  },
}
`;

exports[`Data Handler Should call end with map for info event 1`] = `
Data {
  "blockDevices": FableMap {
    "comparer": Comparer {
      "Compare": [Function],
    },
    "tree": MapTree {
      "data": undefined,
      "tag": 0,
    },
  },
  "zed": ZedData {
    "props": FableSet {
      "comparer": Comparer {
        "Compare": [Function],
      },
      "tree": SetTree {
        "data": undefined,
        "tag": 0,
      },
    },
    "zfs": FableSet {
      "comparer": Comparer {
        "Compare": [Function],
      },
      "tree": SetTree {
        "data": undefined,
        "tag": 0,
      },
    },
    "zpools": FableMap {
      "comparer": Comparer {
        "Compare": [Function],
      },
      "tree": MapTree {
        "data": undefined,
        "tag": 0,
      },
    },
  },
}
`;
